subsys/net/l2/ethernet/ethernet.c

```diff
@@ -207,29 +204,31 @@ static void ethernet_mcast_monitor_cb(struct net_if *iface, const struct net_add
 		return;
 	}
 
-	dev = net_if_get_device(iface);
-	api = dev->api;
-
-	if (!(api->get_capabilities(dev) & ETHERNET_HW_FILTERING) || api->set_config == NULL) {
+	if (!(net_eth_get_hw_capabilities(iface) & ETHERNET_HW_FILTERING)) {
 		return;
 	}
 
-	switch (addr->family) {
+	dev = net_if_get_device(iface);
+	api = dev->api;
+
+	if (api && api->set_config) {
```

Comment:
This change is for set_config callback, could you have a separate commit for it.

Reply:
Done